### PR TITLE
[search] Fixed (and accelerated) coords search.

### PR DESCRIPTION
**Pull request by ygorshenin**
https://jira.mail.ru/browse/MAPSME-2339

@mpimenov PTAL

### CODE REVIEW

File: search/ranker.cpp [hunk 1]
```diff
@@ -275,13 +276,14 @@ class PreResult2Maker
 // static
 size_t const Ranker::kBatchSize = 10;
 
-Ranker::Ranker(Index const & index, storage::CountryInfoGetter const & infoGetter,
-               CategoriesHolder const & categories, vector<Suggest> const & suggests,
-               my::Cancellable const & cancellable)
+Ranker::Ranker(Index const & index, Emitter & emitter,
```
Comment (mpimenov) [inline search/ranker.cpp]: emitter after infoGetter, please

File: search/ranker.cpp [hunk 2]
```diff
@@ -1,3 +1,4 @@
+#include "search/emitter.hpp"
```
Comment (mpimenov) [inline search/ranker.cpp]: Put ranker.hpp first, please.

File: search/emitter.hpp [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+    m_onResults = onResults;
+    m_results.Clear();
+  }
+
+  inline bool AddResult(Result && res) { return m_results.AddResult(move(res)); }
+  inline void AddResultNoChecks(Result && res) { m_results.AddResultNoChecks(move(res)); }
+
+  inline void Emit()
+  {
+    if (m_onResults)
```
Comment (mpimenov) [inline search/emitter.hpp]:
```
else
   log it
```

Comment (ygorshenin) [inline search/emitter.hpp]: Done

File: search/ranker.cpp (hunk 2 above)
Comment (ygorshenin) [inline search/ranker.cpp]: Done

File: search/ranker.cpp (hunk 1 above)
Comment (ygorshenin) [inline search/ranker.cpp]: Done

**Comment by ygorshenin** [comment]
PTAL

**Comment by mpimenov** [comment]
LGTM

**Pull request merged**